04_pig_latin/pig_latin.rb

```diff
@@ -1 +1,22 @@
-#write your code here
+def translate(sentence)
+  words = sentence.split(" ")
+  pig_latin_words = words.map { |word| translate_word(word) }
+  pig_latin_words.join(" ")
+end
+
+def translate_word(word)
+  capitalized = (word[0] == word[0].upcase)
+  word = word.downcase
+  if word =~ /\A[aeiou]/
+    result = word + "ay"
+  elsif word =~ /\A(qu)/
+    result = word[2..-1] + "quay"
+  elsif word =~ /\A([^aeiou]*qu)/
+    match = word.match(/\A([^aeiou]*qu)(.*)/)
+    result = match[2] + match[1] + "ay"
```

Comment:
Using `=~` and then `.match` with almost the same regex immediately after is a bit wasteful because you can already get the match information you need from the `=~` because you can retrieve the capturing groups with `Regexp.last_match` or `$` like so:

```suggestion
  elsif word =~ /\A([^aeiou]*qu)(.*)/
    result = Regexp.last_match(2) + Regexp.last_match(1) + "ay"
```

and the global variable version:

```suggestion
  elsif word =~ /\A([^aeiou]*qu)(.*)/
    result = $2 + $1 + "ay"
```

But it's sometimes unclear that `=~` interacts with `Regexp.last_match`, and global variables are generally frowned upon, so another way is to use the result of #match in the condition like so:

```suggestion
  elsif (match = word.match(/\A([^aeiou]*qu)(.*)/)
    result = match[2] + match[1] + "ay"
```